### PR TITLE
Check and generate docs for updated provision command

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Checked and improved `kyma provision gcp` command.  Added gcp-related information.
- Checked and improved `kyma provision gardener` command.  Added gardener-related information.
- Overridden the main `help`  flag to make it consistent with other flags.
- Improved readme - added a table with commands and child commands, including descriptions and examples.

**Related issue(s)**
See also #276

### CODE REVIEW

File: cmd/kyma/provision/gardener/cmd.go [hunk 1]
```diff
@@ -139,7 +149,7 @@ func newProvider(o *Options) (*types.Provider, error) {
 		v := strings.Split(e, "=")
 
 		if len(v) != 2 {
-			return p, fmt.Errorf("Wrong format for extra configuration %s. Please provide NAME=VALUE pairs.", e)
```
Comment (suleymanakbas91) [inline cmd/kyma/provision/gardener/cmd.go]: Is this `e` deleted by mistake? If not, please adapt the error message as it has a formatting variable `%s` in it.

Comment (bszwarc) [reply cmd/kyma/provision/gardener/cmd.go]: That would be a mistake, let me fix it.

File: cmd/kyma/install/cmd.go [hunk 2]
```diff
@@ -97,12 +97,15 @@ The standard installation uses the minimal configuration. The system performs th
 	cobraCmd.Flags().StringVarP(&o.Domain, "domain", "d", localDomain, "Specifies the domain used for installation.")
 	cobraCmd.Flags().StringVarP(&o.TLSCert, "tlsCert", "", "", "Specifies the TLS certificate for the domain used for installation.")
 	cobraCmd.Flags().StringVarP(&o.TLSKey, "tlsKey", "", "", "Specifies the TLS key for the domain used for installation.")
-	cobraCmd.Flags().StringVarP(&o.Source, "source", "s", DefaultKymaVersion, "Specifies the installation source. To use the specific release, write kyma install --source=1.3.0. To use the latest master, write kyma install --source=latest. To use the local sources, write kyma install --source=local. To use the remote image, write kyma install --source=user/my-kyma-installer:v1.4.0.")
+	cobraCmd.Flags().StringVarP(&o.Source, "source", "s", DefaultKymaVersion, `Specifies the installation source. 
+	- To use the specific release, write "kyma install --source=1.3.0".
+	- To use the latest master, write "kyma install --source=latest".
+	- To use the local sources, write "kyma install --source=local". 
+	- To use the remote image, write kyma "install --source=user/my-kyma-installer:v1.4.0".`)
```
Comment (clebs) [inline cmd/kyma/install/cmd.go]:
```suggestion
	- To use a custom installer image, write kyma "install --source=user/my-kyma-installer:v1.4.0".`)
```

Comment (bszwarc) [reply cmd/kyma/install/cmd.go]: Done.

File: docs/gen-docs/kyma_provision_gardener.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+## kyma provision gardener
+
+Provisions a Kubernetes cluster using Gardener.
+
+### Synopsis
+
+Use this command to use Gardener to provision Kubernetes clusters for Kyma installation.
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gardener.md]:
```suggestion
Use this command to provision Kubernetes clusters on Gardener for Kyma installation. 
```

File: docs/gen-docs/kyma_provision_gardener.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Provisions a Kubernetes cluster using Gardener.
+
+### Synopsis
+
+Use this command to use Gardener to provision Kubernetes clusters for Kyma installation. 
+To successfully provision a cluster on a cloud provider of your choice, you must first create a service account to pass its details as one of the command parameters. 
+Use the following instructions to create a service account for a selected provider:
+- GCP: Check the roles and create a service account using instructions at https://gardener.cloud/050-tutorials/content/howto/gardener_gcp/
+- AWS: Check the roles and create a service account using instructions at https://gardener.cloud/050-tutorials/content/howto/gardener_aws/ 
+- Azure: Create a service account with a `contributor` role. Use service account details to create a Secret and store it in Gardener.
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gardener.md]:
```suggestion
- Azure: Create a service account with the `contributor` role. Use service account details to create a Secret and store it in Gardener.
```

File: docs/gen-docs/kyma_provision_gardener.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+### Options
+
+```
+      --cidr string              Specifies Gardener Classless Inter-Domain Routing (CIDR) of the cluster. (default "10.250.0.0/19")
+  -c, --credentials string       Specifies the path to the kubeconfig file of the Gardener service account for a target provider. (required)
+      --disk-size int            Specifies the disk size (in GB) of the cluster. (default 30)
+      --disk-type string         Specifies the type of disk to use on the target provider. (default "pd-standard")
+  -e, --extra strings            Specifies one or more arguments as "NAME=VALUE" key-value pairs to configure additional cluster settings. You can use this flag multiple times or enter thet key-value pairs as a comma-separated list.
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gardener.md]:
```suggestion
  -e, --extra strings            Specifies one or more arguments as "NAME=VALUE" key-value pairs to configure additional cluster settings. You can use this flag multiple times or enter the key-value pairs as a comma-separated list.
```

File: docs/gen-docs/kyma_provision_gardener.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  -k, --kube-version string      Specifies the Kubernetes version of the cluster. (default "1.15.4")
+  -n, --name string              Specifies the name of the cluster to provision. (required)
+      --nodes int                Specifies the number of nodes of the cluster. (default 3)
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gardener.md]:
```suggestion
      --nodes int                Specifies the number of cluster nodes. (default 3)
```

File: docs/gen-docs/kyma_provision_gardener.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  -p, --project string           Specifies the name of the Gardener project where you provision the cluster. (required)
+  -r, --region string            Specifies the region of the cluster. (default "europe-west3")
+      --scaler-max int           Specifies the maximum autoscale value of the cluster. (default 4)
+      --scaler-min int           Specifies the minimum autoscale value of the cluster. (default 2)
+  -s, --secret string            Specifies the name of the Gardener secret used to access the target provider. (required)
+      --surge int                Specifies the maximum surge of the cluster. (default 4)
+      --target-provider string   Specifies the cloud provider Gardener will use to create the cluster. (default "gcp")
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gardener.md]:
```suggestion
      --target-provider string   Specifies the cloud provider that Gardener should use to create the cluster. (default "gcp")
```

Comment (kazydek) [reply docs/gen-docs/kyma_provision_gardener.md]: I added "that" as I think the "reading experience" is better then  🤔

File: docs/gen-docs/kyma_provision_gardener.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  -t, --type string              Specifies the type of machine of the cluster. (default "n1-standard-4")
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gardener.md]: type of machine of the cluster?

Comment (kazydek) [reply docs/gen-docs/kyma_provision_gardener.md]: `Specifies the machine type for the cluster.`?

File: docs/gen-docs/kyma_provision_gcp.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+```
+kyma provision gcp [flags]
+```
+
+### Options
+
+```
+  -c, --credentials string    Specifies the path to the GCP service account key file. (required)
+      --disk-size int         Specifies the disk size in GB of the cluster. (default 30)
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gcp.md]: `in GB` in () as ib the previous file?

File: docs/gen-docs/kyma_provision_gcp.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  -k, --kube-version string   Specifies the Kubernetes version of the cluster. (default "1.14")
+  -l, --location string       Specifies the location of the cluster. (default "europe-west3-a")
+  -n, --name string           Specifies the name of the GKE cluster to provision. (required)
+      --nodes int             Specifies the number of nodes of the cluster. (default 3)
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gcp.md]:
```suggestion
      --nodes int             Specifies the number of cluster nodes. (default 3)
```

File: docs/gen-docs/kyma_provision_gcp.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  -p, --project string        Specifies the name of the GCP Project where you provision the GKE cluster. (required)
+  -t, --type string           Specifies the type of machine of the cluster. (default "n1-standard-4")
```
Comment (kazydek) [inline docs/gen-docs/kyma_provision_gcp.md]: as in the previous file

File: docs/gen-docs/kyma_install.md [hunk 12]
```diff
@@ -48,22 +48,26 @@ kyma install [flags]
...
+  -o, --override stringArray   Specifies the path to a YAML file with parameters to override.
+  -p, --password string        Specifies the predefined cluster password.
+  -s, --source string          Specifies the installation source. 
+                               	- To use the specific release, write "kyma install --source=1.3.0".
+                               	- To use the latest master, write "kyma install --source=latest".
+                               	- To use the local sources, write "kyma install --source=local". 
+                               	- To use a custom installer image, write kyma "install --source=user/my-kyma-installer:v1.4.0".
+      --src-path string        Specifies the absolute path to local sources.
+      --timeout duration       Time-out after which CLI stops watching the installation progress. (default 1h0m0s)
+      --tlsCert string         Specifies the TLS certificate for the domain used for installation.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_install.md]: Yes, I removed "the" for Kyma installation.

File: docs/gen-docs/kyma_install.md [hunk 13]
```diff
@@ -48,22 +48,26 @@ kyma install [flags]
...
+      --tlsKey string          Specifies the TLS key for the domain used for installation.
```
[the remark this answers is not in the document]
Comment (bszwarc) [reply docs/gen-docs/kyma_install.md]: 👍

File: docs/gen-docs/kyma_provision_gardener.md (hunk 3 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gardener.md]: 👍

File: docs/gen-docs/kyma_provision_gardener.md (hunk 4 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gardener.md]: 👍

File: docs/gen-docs/kyma_provision_gardener.md (hunk 5 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gardener.md]: 👍

File: docs/gen-docs/kyma_provision_gardener.md (hunk 6 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gardener.md]: 👍

File: docs/gen-docs/kyma_provision_gardener.md (hunk 7 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gardener.md]: Yes, you are right.

File: docs/gen-docs/kyma_provision_gardener.md (hunk 8 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gardener.md]: Corrected.

File: docs/gen-docs/kyma_provision_gcp.md (hunk 9 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gcp.md]: 👍

File: docs/gen-docs/kyma_provision_gcp.md (hunk 10 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gcp.md]: 👍

File: docs/gen-docs/kyma_provision_gcp.md (hunk 11 above)
Comment (bszwarc) [reply docs/gen-docs/kyma_provision_gcp.md]: Done.

**Pull request merged**